front/lib/upsert_document.ts

```diff
@@ -37,44 +38,107 @@ export const EnqueueUpsertDocument = t.type({
   upsertContext: t.union([UpsertContextSchema, t.null]),
 });
 
+export const EnqueueUpsertTable = t.type({
+  workspaceId: t.string,
+  projectId: t.string,
+  dataSourceName: t.string,
+  tableName: t.string,
+  tableDescription: t.string,
+  tableId: t.string,
+  csv: t.union([t.string, t.null]),
+  truncate: t.boolean,
+});
+
+type EnqueueUpsertDocumentType = t.TypeOf<typeof EnqueueUpsertDocument>;
+
+type EnqueueUpsertTableType = t.TypeOf<typeof EnqueueUpsertTable>;
+
 export async function enqueueUpsertDocument({
   upsertDocument,
 }: {
-  upsertDocument: t.TypeOf<typeof EnqueueUpsertDocument>;
+  upsertDocument: EnqueueUpsertDocumentType;
+}): Promise<Result<string, Error>> {
+  const upsertQueueId = uuidv4();
+
+  logger.info(
+    {
+      upsertQueueId,
+      workspaceId: upsertDocument.workspaceId,
+      dataSourceName: upsertDocument.dataSourceName,
+      documentId: upsertDocument.documentId,
+      enqueueTimestamp: Date.now(),
+    },
+    "[UpsertQueue] Enqueueing document"
+  );
+
+  return enqueueUpsert({
+    upsertItem: upsertDocument,
+    upsertQueueId,
+    launchWorkflowFn: launchUpsertDocumentWorkflow,
+  });
+}
+
+export async function enqueueUpsertTable({
+  upsertTable,
+}: {
+  upsertTable: EnqueueUpsertTableType;
 }): Promise<Result<string, Error>> {
+  const upsertQueueId = uuidv4();
+
+  logger.info(
+    {
+      upsertQueueId,
+      workspaceId: upsertTable.workspaceId,
+      dataSourceName: upsertTable.dataSourceName,
+      documentId: upsertTable.tableId,
+      enqueueTimestamp: Date.now(),
+    },
+    "[UpsertQueue] Enqueueing table"
+  );
+
+  return enqueueUpsert({
+    upsertItem: upsertTable,
+    upsertQueueId,
+    launchWorkflowFn: launchUpsertDocumentWorkflow,
+  });
+}
+
+async function enqueueUpsert({
+  upsertItem,
+  upsertQueueId,
+  launchWorkflowFn,
+}:
+  | {
+      upsertItem: t.TypeOf<typeof EnqueueUpsertDocument>;
```

Comment:
Let's use the right type here 😉 
```suggestion
      upsertItem: EnqueueUpsertDocumentType;
```